packages/devtools_app/lib/src/screens/memory/shared/heap/heap.dart

```diff
@@ -29,17 +35,62 @@ class AdaptedHeap {
       singleHeapClass.countInstance(data, i);
     }
 
-    return SingleHeapClasses(result)..seal();
+    return SingleHeapClasses(result, filter)..seal();
   }
 }
 
-abstract class HeapClasses with Sealable {
-  Iterable<ClassStats> get classStatsList;
+abstract class HeapClasses<T extends ClassStats> with Sealable {
+  HeapClasses(this.filter);
+
+  List<T> get classStatsList;
+
+  final ValueListenable<ClassFilter> filter;
+}
+
+mixin Filterable<T extends ClassStats> on HeapClasses<T> {
```

Comment:
we should try to reuse the existing filter framework in devtools instead of reinventing the wheel here. Look at `filter.dart`. We should be able to use `FilterControllerMixin`

Reply:
I reviewed it. It makes sense to reuse UI, but, as there is a lot of specifics in diffing, it seems reuse of data processing will bring more complexity, than value.

Here is refactoring needed to reuse UI: https://github.com/flutter/devtools/pull/4638